### PR TITLE
Tweak dialogue codegen

### PR DESCRIPTION
**Pull request by iamdanfox**
## Before this PR

A given conjure FooService would generate:

- DialogueFooService (with a bunch of endpoints, and an `async()` and `blocking()`)
- BlockingFooService
- AsyncFooService

This turned out a little tricky to use from the Witchcraft facade, and required a bit of reflection. We tried a few different options for the codegen, including using a `@ConstructUsing` annotation and some enums.

## After this PR
==COMMIT_MSG==
For a given conjure `FooService`, the static factory methods to create blocking and async versions are now on the `BlockingFooService` and `AsyncFooService`.
==COMMIT_MSG==

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+**/integrationInput/** linguist-generated=true
```
Comment (ferozco) [inline .gitattributes:1]: nice!

Comment (ferozco) [reply .gitattributes:1]: might want to add test resources too

File: conjure-java-core/src/test/resources/test/api/DialogueCookieEndpoints.java.dialogue [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                }
+
+                @Override
+                public String version() {
+                    return "";
+                }
+            };
+
+    private DialogueCookieEndpoints() {}
+}
```
Comment (iamdanfox) [inline conjure-java-core/src/test/resources/test/api/DialogueCookieEndpoints.java.dialogue:42]: in this endpoints class everything can be package-private

File: conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/DialogueEndpointsGenerator.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    }
+
+    public JavaFile endpointsClass(ServiceDefinition def) {
+        ClassName serviceClassName = Names.endpointsClassName(def, options);
+
+        TypeSpec endpointsClass = TypeSpec.classBuilder(serviceClassName)
+                .addModifiers(Modifier.FINAL)
+                .addAnnotation(ConjureAnnotations.getConjureGeneratedAnnotation(DialogueEndpointsGenerator.class))
+                .addFields(def.getEndpoints().stream()
+                        .map(e -> endpointField(e, def.getServiceName().getName(), apiVersion))
```
Comment (ferozco) [inline conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/DialogueEndpointsGenerator.java]: nit: s/e/endpoint

File: conjure-java-core/src/integrationInput/java/com/palantir/product/DialogueEteEndpoints.java [hunk 4]
```diff
@@ -0,0 +1,918 @@
...
+import com.palantir.dialogue.HttpMethod;
+import com.palantir.dialogue.PathTemplate;
+import com.palantir.dialogue.UrlBuilder;
+import java.lang.Override;
+import java.lang.String;
+import java.util.Map;
+import javax.annotation.Generated;
+
+@Generated("com.palantir.conjure.java.services.dialogue.DialogueEndpointsGenerator")
+final class DialogueEteEndpoints {
```
Comment (carterkozak) [inline conjure-java-core/src/integrationInput/java/com/palantir/product/DialogueEteEndpoints.java]: Should this be an enum?

```java
enum DialogueEteEndpoints extends Endpoint {
    STRING() {
    /* etc */
}
```

**Review by ferozco — COMMENTED** [review]
Looks good

File: conjure-java-core/src/integrationInput/java/com/palantir/product/DialogueEteEndpoints.java (hunk 4 above)
Comment (iamdanfox) [reply conjure-java-core/src/integrationInput/java/com/palantir/product/DialogueEteEndpoints.java]: oh that's neat - will update

**Comment by iamdanfox** [comment]
Updated and ready for another look :)

**Comment by ferozco** [comment]
👍

**Comment by iamdanfox** [comment]
🐌

**Pull request merged**